src/ucs/sys/compiler_def.h

```diff
@@ -135,7 +135,7 @@
  * @return Address of the container structure.
  */
 #define ucs_container_of(_ptr, _type, _member) \
-    ( (_type*)( (char*)(void*)(_ptr) - ucs_offsetof(_type, _member) )  )
+    ( (_type*)( (uintptr_t)(void*)(_ptr) - ucs_offsetof(_type, _member) )  )
```

Comment:
why needed?

Reply:
GCC10 fails to compile, a warning complaining on ptr[-2] indexing to an array.

Reply:
so this fix is not really related to "Ptr array foreach improve by dcache prefetching"?

Reply:
Not related to the same pull request, but I figured it can be part of the same PR...

Reply:
To clarify: this also fixes a (new...) GCC v10 issue shuki found while testing the optimization.

BTW @shuki-zanyovka - according to the CI this is a problem on other GCCs... need to change this to something all these GCC-s will like :)

Reply:
Closing this PR, I will create 2 separate PRs for the 2 issues.